rocAL/include/pipeline/tensor.h

```diff
@@ -196,8 +196,14 @@ class TensorInfo {
             get_modified_dims_from_layout(_layout, layout, new_dims);
             _dims = new_dims;
             modify_strides();
+            _max_shape.assign(_dims.begin() + 1, _dims.end());
```

Comment:
Set max shape does this and also the channels value assignment
Why do we need to do here?